src/test/lib/ApiTest.cpp

```diff
@@ -1753,8 +1753,210 @@ class SecConfigTestContext {
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+
+    {
+        QUIC_SETTINGS Settings{0};
+        Settings.HandshakeIdleTimeoutMs = UINT64_MAX;
+        Settings.IsSet.HandshakeIdleTimeoutMs = TRUE;
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+
+    {
+        QUIC_SETTINGS Settings{0};
+        Settings.StreamRecvBufferDefault = 0;
+        Settings.IsSet.StreamRecvBufferDefault = TRUE;
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+
+    {
+        QUIC_SETTINGS Settings{0};
+        Settings.MaxBytesPerKey = UINT64_MAX;
+        Settings.IsSet.MaxBytesPerKey = TRUE;
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+
+    {
+        QUIC_SETTINGS Settings{0};
+        Settings.ServerResumptionLevel = 3; // bit length 2
+        Settings.IsSet.ServerResumptionLevel = TRUE;
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+
+    //
+    // MinimumMtu is bigger than MaximumMtu
+    //
+    {
+        QUIC_SETTINGS Settings{0};
+        Settings.MinimumMtu = 1400;
+        Settings.IsSet.MinimumMtu = TRUE;
+        Settings.MaximumMtu = 1300;
+        Settings.IsSet.MaximumMtu = TRUE;
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+
+    //
+    // Good
+    //
+    {
+        QUIC_SETTINGS Settings{0};
+
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->SetParam(
+                Handle,
+                Param,
+                sizeof(QUIC_SETTINGS),
+                &Settings));
+    }
+}
+
+void QuicTestStatefulGlobalSetParam()
+{
+    TestScopeLogger LogScope("QuicTestStatefulGlobalSetParam");
+    MsQuicRegistration Registration(true);
+    TEST_TRUE(Registration.IsValid());
+
+    //
+    // Set QUIC_PARAM_GLOBAL_LOAD_BALACING_MODE after connection start (MsQuicLib.InUse)
+    //
+    {
+        TestScopeLogger LogScope("Set QUIC_PARAM_GLOBAL_LOAD_BALACING_MODE after connection start (MsQuicLib.InUse)");
+        GlobalSettingScope ParamScope(QUIC_PARAM_GLOBAL_LOAD_BALACING_MODE);
```

Comment:
You should only need this helper if you plan to successfully set the parameter, right?

Reply:
Yes!